models/issue.go

```diff
@@ -1498,3 +1507,46 @@ func updateIssue(e Engine, issue *Issue) error {
 func UpdateIssue(issue *Issue) error {
 	return updateIssue(x, issue)
 }
+
+// UpdateIssueDeadline does what it says
+func UpdateIssueDeadline(issue *Issue, doer *User) (err error) {
```

Comment:
You can change this to `func UpdateIssueDeadline(issue *Issue, DeadlineUnix util.TimeStamp, doer *User) error {`. Then you can remove `actualIssue`.

Reply:
But wouldn't that mean to get the actual deadline before calling the function?

Reply:
Maybe something like this?
```
	var deadline util.TimeStamp
	if form.Deadline != nil && !form.Deadline.IsZero() {
		deadline  = util.TimeStamp(form.Deadline.Unix())
	}
	if err := models.UpdateIssueDeadline(issue, deadline, ctx.User); err != nil {
		ctx.Error(500, "UpdateIssueDeadline", err)
	}
```

Reply:
But I'd still need to check if the issue deadline hasn't changed meanwhile.